### PR TITLE
Added missing fields in the google_data_loss_prevention_deidentify_template resource

### PR DESCRIPTION
**Pull request by JayS-crest**
Added `image_transformations`, `replace_dictionary_config`, `version`, `create_time` and `update_time` to `data_loss_prevention_deidentify_template`

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
dlp: added `image_transformations` field to `google_data_loss_prevention_deidentify_template` resource
```
```release-note:enhancement
dlp: added `replace_dictionary_config` field to `info_type_transformations` in `google_data_loss_prevention_deidentify_template` resource
```

### CODE REVIEW

**Comment by zli82016** [comment]
> Hi, actually I tried viewing `presubmit-generate-diff` details but in the console, I've received an error stating `You do not have sufficient permissions to view this page`. Could you help me with the error please, Thanks!

I found this error

/mm-tf-oics-beta-head/mmv1/google/yaml_validator.rb:64:in `check': deidentify_config.info_type_transformations.transformations.transformations.primitive_transformation.replace_dictionary_config.word_list.words > Missing 'item_type' (RuntimeError)

**Comment by JayS-crest** [comment]
> > Hi, actually I tried viewing `presubmit-generate-diff` details but in the console, I've received an error stating `You do not have sufficient permissions to view this page`. Could you help me with the error please, Thanks!
> 
> I found this error
> 
> /mm-tf-oics-beta-head/mmv1/google/yaml_validator.rb:64:in `check': deidentify_config.info_type_transformations.transformations.transformations.primitive_transformation.replace_dictionary_config.word_list.words > Missing 'item_type' (RuntimeError)

Thanks for the help! I missed out on mentioning `item_type` for an attribute that happens to be array. I've done the necessary changes.

File: mmv1/products/dlp/DeidentifyTemplate.yaml [hunk 1]
```diff
@@ -1573,6 +1692,12 @@ properties:
                               description: |
                                 Words or phrases defining the dictionary. The dictionary must contain at least one phrase and every phrase must contain at least 2 characters that are letters or digits.
                               item_type: Api::Type::String
+                    - !ruby/object:Api::Type::Boolean
+                      name: 'replaceWithInfoTypeConfig'
```
Comment (JayS-crest) [inline mmv1/products/dlp/DeidentifyTemplate.yaml:1696]: While running the tests, I saw error for boolean and tried to pass empty object, that worked but further I received API Error stating `Error creating DeidentifyTemplate: googleapi: Error 400: replace_with_info_type_config is not valid as a primitive_transformations. It is only compatible with info_type_transformations.`

Hence, removing this field and respective added test case.

**Comment by JayS-crest** [comment]
Hi, I've done the necessary changes and it seems like the tests should work now.

File: mmv1/products/dlp/DeidentifyTemplate.yaml [hunk 2]
```diff
@@ -68,18 +73,110 @@ properties:
...
+                      default_value: 0.0
+                    - !ruby/object:Api::Type::Double
+                      name: green
+                      description: |
+                        The amount of green in the color as a value in the interval [0, 1].
+                      default_value: 0.0
+                - !ruby/object:Api::Type::NestedObject
+                  name: 'selectedInfoTypes'
+                  description: Apply transformation to the selected infoTypes.
+                  exactly_one_of:
```
Comment (zli82016) [inline mmv1/products/dlp/DeidentifyTemplate.yaml:129]: `exactly_one_of` does not for the list and generates `ExactlyOneOf: []string{}` in the downstream provider.

You can remove it. I think the service side handle should handle it already.

File: mmv1/products/dlp/DeidentifyTemplate.yaml [hunk 3]
```diff
@@ -68,18 +73,110 @@ properties:
...
+                          description: |
+                            Name of the information type.
+                        - !ruby/object:Api::Type::String
+                          name: 'version'
+                          description: |
+                            Version name for this InfoType.
+                - !ruby/object:Api::Type::NestedObject
+                  name: 'allInfoTypes'
+                  description: Apply transformation to all findings not specified in other ImageTransformation's selectedInfoTypes.
+                  exactly_one_of:
```
Comment (zli82016) [inline mmv1/products/dlp/DeidentifyTemplate.yaml:153]: `exactly_one_of` can be removed.

File: mmv1/products/dlp/DeidentifyTemplate.yaml [hunk 4]
```diff
@@ -68,18 +73,110 @@ properties:
...
+                    - transforms.0.all_info_types
+                    - transforms.0.all_text
+                  properties: [] # Meant to be an empty object with no properties - see here : https://cloud.google.com/dlp/docs/reference/rest/v2/projects.deidentifyTemplates#allinfotypes
+                    # The fields below are necessary to include the "allInfoTypes" transformation in the payload
+                  send_empty_value: true
+                  allow_empty_object: true
+                - !ruby/object:Api::Type::NestedObject
+                  name: 'allText'
+                  description: Apply transformation to all text that doesn't match an infoType.
+                  exactly_one_of:
```
Comment (zli82016) [inline mmv1/products/dlp/DeidentifyTemplate.yaml:164]: exactly_one_of can be removed.

File: mmv1/products/dlp/DeidentifyTemplate.yaml [hunk 5]
```diff
@@ -68,18 +73,110 @@ properties:
...
+                - !ruby/object:Api::Type::NestedObject
+                  name: 'selectedInfoTypes'
+                  description: Apply transformation to the selected infoTypes.
+                  exactly_one_of:
+                    - transforms.0.selected_info_types
+                    - transforms.0.all_info_types
+                    - transforms.0.all_text
+                  properties:
+                  - !ruby/object:Api::Type::Array
+                    name: 'infoTypes'
```
Comment (zli82016) [inline mmv1/products/dlp/DeidentifyTemplate.yaml:135]: `infoTypes` can be required, as it is the only filed inside the object `selectedInfoTypes`

File: mmv1/products/dlp/DeidentifyTemplate.yaml [hunk 6]
```diff
@@ -450,6 +551,23 @@ properties:
                           name: 'radix'
                           description: |
                             The native way to select the alphabet. Must be in the range \[2, 95\].
+                    - !ruby/object:Api::Type::NestedObject
+                      name: 'replaceDictionaryConfig'
```
Comment (zli82016) [inline mmv1/products/dlp/DeidentifyTemplate.yaml:555]: Can you please add a test example for this field? Thanks.

File: mmv1/products/dlp/DeidentifyTemplate.yaml (hunk 5 above)
Comment (JayS-crest) [reply mmv1/products/dlp/DeidentifyTemplate.yaml:131]: Added `required:true` to `infoTypes`.

File: mmv1/products/dlp/DeidentifyTemplate.yaml (hunk 2 above)
Comment (JayS-crest) [reply mmv1/products/dlp/DeidentifyTemplate.yaml]: Removed `exactlyOneOf` from all the three fields.

File: mmv1/products/dlp/DeidentifyTemplate.yaml (hunk 3 above)
Comment (JayS-crest) [reply mmv1/products/dlp/DeidentifyTemplate.yaml]: Removed `exactlyOneOf` from all the three fields.

File: mmv1/products/dlp/DeidentifyTemplate.yaml (hunk 4 above)
Comment (JayS-crest) [reply mmv1/products/dlp/DeidentifyTemplate.yaml]: Removed `exactlyOneOf` from all the three fields.

File: mmv1/products/dlp/DeidentifyTemplate.yaml (hunk 6 above)
Comment (JayS-crest) [reply mmv1/products/dlp/DeidentifyTemplate.yaml:544]: Actually, I've already added create-update test of this field in an existing test function namely  `TestAccDataLossPreventionDeidentifyTemplate_dlpDeidentifyTemplate_infoTypeTransformationsUpdate`. The config where it is updated can be found [here](https://github.com/GoogleCloudPlatform/magic-modules/pull/7816/files#diff-ec31eff9a8a9a675959b0a56dbf6b1b45df815865408a2cad9a9bb4c6026e451R240) for create and [here](https://github.com/GoogleCloudPlatform/magic-modules/pull/7816/files#diff-ec31eff9a8a9a675959b0a56dbf6b1b45df815865408a2cad9a9bb4c6026e451R409) for update.

**Review by zli82016 — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**